src/picongpu/include/particles/manipulators/RandomPositionImpl.def

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2015 Rene Widera
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+
+
+/** Change the in cell position
+ *
+ * This functor change the in cell position of all particle of a species
+ *
+ * @tparam T_SpeciesType type of the species which should be manipulated
+ *
+ * example: add
+ *   `particles::Manipulate<particles::manipulators::RandomPositionImpl<>,SPECIES_NAME>`
```

Comment:
we might better add them as documentation in above the initialization pipeline in the according param file